lib/mcp/mcp.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mcp
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"sync"
+
+	"github.com/gravitational/trace"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+)
+
+// Tool defines an interface an MCP tool being registered with Teleport MCP
+// server must implement.
+type Tool interface {
+	// GetTool returns the MCP tool definition.
+	GetTool() mcp.Tool
+	// GetHandler returns the MCP tool handler.
+	GetHandler() server.ToolHandlerFunc
+}
+
+type toolMaker func(cfg Config) (Tool, error)
+
+var mu sync.Mutex
+var toolsRegistry map[string][]toolMaker = make(map[string][]toolMaker)
+
+// RegisterTool adds an MCP tool to the registry of available tools for the
+// server to register.
+func RegisterTool(toolset string, t toolMaker) {
+	mu.Lock()
+	defer mu.Unlock()
+	toolsRegistry[toolset] = append(toolsRegistry[toolset], t)
```

Comment:
should the tools be only registered when feature is enabled for the cluster?

is a global `toolsRegistry` necessary? (could the MCP server be created on the fly without it?)